types/src/byzantine.rs

```diff
@@ -110,25 +110,74 @@ impl FraudProof for BadEncodingFraudProof {
 
         // verify that Merkle proofs correspond to particular shares.
         for share in &self.shares {
+            let Some(share) = share else {
+                continue;
+            };
             share
                 .proof
-                .verify_range(
-                    &root,
-                    &[share.leaf.share.as_ref()],
-                    share.leaf.namespace.into(),
-                )
+                .verify_range(&root, &[&share.leaf.share], share.leaf.namespace.into())
                 .map_err(Error::RangeProofError)?;
         }
 
-        // TODO: Add leopard reed solomon decoding and encoding of shares
-        //       and verify the nmt roots.
+        // rebuild the whole axis
+        let mut rebuilt_shares: Vec<_> = self
+            .shares
+            .iter()
+            .map(|share| {
+                share
+                    .as_ref()
+                    .map(|sh| sh.leaf.share.to_vec())
+                    .unwrap_or_default()
+            })
+            .collect();
+        let data_shares = rebuilt_shares.len() / 2;
```

Comment:
I find `*_shares` variables containing shares in one place and number of shares in another slightly confusing to read. Could we do `data_shares_number` here?

Reply:
how about now?